### PR TITLE
Prep for v4.0

### PR DESCRIPTION
**Pull request by r-devulap**

### CODE REVIEW

File: src/README.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+void avx512_qsort_kv<T>(T1* key, T2* value , size_t arrsize)
+```
+Supported datatypes: `uint64_t, int64_t and double`
+
+## Algorithm details
+
+The ideas and code are based on these two research papers [1] and [2]. On a
+high level, the idea is to vectorize quicksort partitioning using AVX-512
+compressstore instructions. If the array size is < 512, then use Bitonic
+sorting network implemented on 512-bit registers.  The precise network
```
Comment (sterrettm2) [inline src/README.md:96]: This is no longer accurate

File: src/README.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+partialsort and AVX-512 implementations of argsort, argselect and key-value
+sort. The following API's are currently supported:
+
+#### Quicksort
+
+Equivalent to `qsort` in
+[C](https://www.tutorialspoint.com/c_standard_library/c_function_qsort.htm) or
+`std::sort` in [C++](https://en.cppreference.com/w/cpp/algorithm/sort).
+
+```cpp
```
Comment (sterrettm2) [inline src/README.md:15]: I think this should mention the hasnan parameter and its behavior

**Review by sterrettm2 — COMMENTED** [review]
Looks good, these are the only small things I've noticed

File: src/README.md (hunk 2 above)
Comment (r-devulap) [reply src/README.md:15]: good catch, fixed it.

File: src/README.md (hunk 1 above)
Comment (r-devulap) [reply src/README.md:96]: fixed it

**Pull request merged**